lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Http/Monitor/API.hs

```diff
@@ -91,9 +160,69 @@ parseHistory = withArray "History" $ \arr -> do
   where
     parsePhase = withObject "Phase" $ \o -> do
         time <- o .: "time"
-        phase <- o .: "phase"
+        phase <- o .: "phase" >>= parsePhase'
         pure (time, phase)
 
+type Tags a = [(Text, [Either a (Value -> Parser a)])]
+
+pattern T :: a -> b -> (a, [b])
+pattern T a b = (a, [b])
+
+pattern Ts :: a -> [b] -> (a, [b])
+pattern Ts a bs = (a, bs)
```

Comment:
```suggestion
tag :: a -> b -> (a, [b])
tag a b = (a, [b])

tags :: a -> [b] -> (a, [b])
tags a bs = (a, bs)
```

As you never match any of these patters, traditional smart constructors would also be an option. I leave it up to you — `PatternSynonyms` is an additional language extension that is not strictly required, but it's a language extension that I like, so I would be ok with it.